opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -164,38 +193,42 @@ namespace Opm {
 
 
     template<typename TypeTag>
-    SimulatorReport
+    void
     BlackoilWellModel<TypeTag>::
-    assemble(Simulator& ebosSimulator,
-             const int iterationIdx,
-             const double dt,
-             WellState& well_state)
+    assemble(const int iterationIdx,
+             const double dt)
     {
-        SimulatorReport report;
-        if ( ! wellsActive() ) {
-            return report;
-        }
+
+
+        last_report_ = SimulatorReport();
+
+        updatePerforationIntensiveQuantities();
 
         if (iterationIdx == 0) {
-            prepareTimeStep(ebosSimulator, well_state);
+            prepareTimeStep();
+        }
+
+
+
+        if ( ! wellsActive() ) {
+            return;
```

Comment:
Why is this moved down? It would be better to have it before the calls to updatePerforationIntensiveQuantities() and prepareTimeStep().